var/spack/repos/builtin/packages/mvapich-plus/package.py

```diff
@@ -0,0 +1,365 @@
+    )
+
+    variant(
+        "file_systems",
+        description="List of the ROMIO file systems to activate",
+        values=auto_or_any_combination_of("lustre", "gpfs", "nfs", "ufs"),
+    )
+
+    depends_on("findutils", type="build")
+    depends_on("bison", type="build")
+    depends_on("pkgconfig", type="build")
+    depends_on("zlib-api")
+    depends_on("libpciaccess", when=(sys.platform != "darwin"))
+    depends_on("libxml2")
+    depends_on("cuda", when="+cuda", type=("build", "link"))
+    depends_on("libfabric", when="netmod=ofi")
+    depends_on("slurm", when="process_managers=slurm")
+    depends_on("ucx", when="netmod=ucx")
+    depends_on("hip@3.9.0:", when="+rocm")
+    depends_on("rccl", when="+rocm")
+    variant("rocm", description="Enable/Disable support for ROCM", default=False)
+    conflicts("+cuda +rocm", msg="MVAPICH2-GDR can only be built with either CUDA or ROCm")
+    conflicts("~cuda ~rocm", msg="MVAPICH2-GDR must be built with either CUDA or ROCm")
+    conflicts(
+        "process_managers=slurm pmi_version=simple",
+        msg="MVAPICH2-GDR can not be built with slurm and simple pmi",
+    )
+
+    with when("process_managers=slurm"):
+        conflicts("pmi_version=pmi2")
+
+    with when("process_managers=auto"):
+        conflicts("pmi_version=pmi2")
+
+    filter_compiler_wrappers("mpicc", "mpicxx", "mpif77", "mpif90", "mpifort", relative_root="bin")
+
+    @classmethod
+    def determine_version(cls, exe):
+        if exe.endswith("mpichversion"):
+            output = Executable(exe)(output=str, error=str)
+            match = re.search(r"^MVAPICH2 Version:\s*(\S+)", output)
+        elif exe.endswith("mpiname"):
+            output = Executable(exe)("-a", output=str, error=str)
+            match = re.search(r"^MVAPICH2 (\S+)", output)
+        return match.group(1) if match else None
+
+    @property
+    def libs(self):
+        query_parameters = self.spec.last_query.extra_parameters
+        libraries = ["libmpi"]
+
+        if "cxx" in query_parameters:
+            libraries = ["libmpicxx"] + libraries
+
+        return find_libraries(libraries, root=self.prefix, shared=True, recursive=True)
+
+    @property
+    def process_manager_options(self):
+        spec = self.spec
+
+        other_pms = []
+        for x in ("hydra", "gforker", "remshell"):
+            if "process_managers={0}".format(x) in spec:
+                other_pms.append(x)
+
+        opts = []
+        if len(other_pms) > 0:
+            opts = ["--with-pm=%s" % ":".join(other_pms)]
+
+        # See: http://slurm.schedmd.com/mpi_guide.html#mvapich2
+        if "process_managers=slurm" in spec:
+            opts = [
+                "--with-pm=slurm",
+                #        "--with-pmi=simple",
+                "--with-slurm={0}".format(spec["slurm"].prefix),
+                "CFLAGS=-I{0}/include/slurm".format(spec["slurm"].prefix),
+            ]
```

Comment:
It's preferable to use `spec.satisfies()` instead of `x in spec` since the latter can have unintended consequences.